src/test/java/emissary/util/PayloadUtilTest.java

```diff
@@ -98,10 +98,60 @@ void testMultiLineString() {
+        assertTrue(answer.contains("BAR.UNKNOWN.BARPLACE.http://example.com:1234/BarPlace"),
+                "Answer did not contain the correct transform history entry");
+    }
+
+    @Test
+    void testReducedHistory() {
+        // setup
+        final String fn = "testReducedHistory";
+        final IBaseDataObject d = DataObjectFactory.getInstance("abc".getBytes(), fn, Form.UNKNOWN);
+        d.appendTransformHistory("FOO.UNKNOWN.FOOPLACE.http://example.com:1234/FooPlace");
+        d.appendTransformHistory("BAR.UNKNOWN.BARPLACE.http://example.com:1234/BarPlace");
+        d.appendTransformHistory("TEST.DROPOFFPLACE.http://example.com:1234/DropOffPlace");
+        d.setCreationTimestamp(new Date(0));
+
+        // test
+        PayloadUtil.reducedTransformHistory.add("testReducedHistory");
+        final String answer = PayloadUtil.getPayloadDisplayString(d, false);
+        PayloadUtil.reducedTransformHistory.clear();
+
+        // verify
+        assertTrue(answer.contains("\n"), "Must be multi-line string");
+        assertTrue(answer.contains("filename: testReducedHistory"), "Answer did not contain the correct filename");
+        assertTrue(answer.contains("transform history (3)"), "Answer did not contain the transform history number");
+        assertFalse(answer.contains("FOO.UNKNOWN.FOOPLACE.http://example.com:1234/FooPlace"),
+                "Answer should not contain this transform history entry");
+        assertFalse(answer.contains("BAR.UNKNOWN.BARPLACE.http://example.com:1234/BarPlace"),
+                "Answer should not contain this transform history entry");
+        assertTrue(answer.contains("** reduced transform history **"), "Answer should contain 'reduced transform history'");
+        assertTrue(answer.contains("dropOff -> TEST.DROPOFFPLACE.http://example.com:1234/DropOffPlace"), "Answer should show dropoff");
+    }
+
+    @Test
+    void testNoUrlHistory() {
+        // setup
+        final String fn = "noUrlHistory";
+        final IBaseDataObject d = DataObjectFactory.getInstance("abc".getBytes(), fn, Form.UNKNOWN);
+        d.appendTransformHistory("FOO.UNKNOWN.FOOPLACE.http://example.com:1234/FooPlace");
+        d.appendTransformHistory("BAR.UNKNOWN.BARPLACE.http://example.com:1234/BarPlace");
+        d.setCreationTimestamp(new Date(0));
+
+        // test
+        PayloadUtil.noURLHistory.add("noUrlHistory");
+        final String answer = PayloadUtil.getPayloadDisplayString(d, false);
+        PayloadUtil.noURLHistory.clear();
+
+        // verify
+        assertTrue(answer.contains("\n"), "Must be multi-line string");
+        assertTrue(answer.contains("filename: noUrlHistory"), "Answer did not contain the correct filename");
+        assertTrue(answer.contains("FOO.UNKNOWN.FOOPLACE"),
+                "Answer should not contain the URL");
+        assertTrue(answer.contains("BAR.UNKNOWN.BARPLACE"),
+                "Answer should not contain the URL");
```

Comment:
In addition to the above two, you can also add these two statements to confirm that the full URL is not present
`assertFalse(answer.contains("FOO.UNKNOWN.FOOPLACE.http://example.com:1234/FooPlace"), ...`
`assertFalse(answer.contains("BAR.UNKNOWN.BARPLACE.http://example.com:1234/BarPlace"), ...`